Lib/os.py

```diff
@@ -715,15 +715,27 @@ def setdefault(self, key, value):
 try:
     _putenv = putenv
 except NameError:
-    _putenv = lambda key, value: None
+    def _putenv(key, value):
+        # do nothing
+        return
 else:
     if "putenv" not in __all__:
         __all__.append("putenv")
 
 try:
     _unsetenv = unsetenv
 except NameError:
-    _unsetenv = lambda key: _putenv(key, "")
+    if name == 'nt':
```

Comment:
This looks too complicated and obscure. Many lines are duplicated with small differences. Would not it be easier to implement `nt.unsetenv()` on C?